### PR TITLE
SWEP changes and some missile changes

### PR DESCRIPTION
**Pull request by RedDeadlyCreeper**
-Improved IR missile flare registration. Gave IR missiles increased weight for the current target.
-Fixed LOS trace for laser guidance
-Made flares able to be detected by radar "Chaff," increased ground clutter distance
-Added GPS guidance, a one time positional guidance that guides missiles to a position regardless of the lock cone or distance
-Adjusted temperature of flares now that the heat is properly registered
-Cleaned up SWEP missile lead, gravity compensation, and generalized the functions. Improved guidance, detonation, and tracking.
-SWEP missiles can now be flared.
-Nerfed stinger energy and blast radius
-remove IR from most ASMs and ATGMs, Javelin can keep it but gets a narrower cone
-Remove Position output from IRST
-Re-added restart delay after changing IRST cone
-Reduced Plunging Fuse penetration energy
-Added sight to XM25 with drop calculations (Fuse needs to account for drop)
-Added aim to glock 25
-Increased the ammo of most sweps to 1 + 5 magazines

### CODE REVIEW

File: lua/acf/shared/fuses/f_plunging.lua [hunk 1]
```diff
@@ -84,16 +84,19 @@ end
...
 function this:PerformDetonation( missile, bdata, phys, pos )
 
-	--HEAT system breaks and it becomes unusable. THEAT cannot be used until a proof can justify it.
+	--HEAT system breaks and it becomes unusable. THEAT cannot be used until a proof can justify it. Really tandem doesn't exist since it's an Explosively formed penetrator.
 	if not missile.PlungingDetonation or bdata.Type == "THEAT" then
 		bdata.Type = "HE"
 	end
 
-	bdata.Flight = missile:GetUp() * - 100
+	bdata.Flight = missile:GetUp() * - 200
```
Comment (KemGus) [inline lua/acf/shared/fuses/f_plunging.lua:92]: Magic number. It is somewhat understandable, but would be nice to make like

Comment (KemGus) [reply lua/acf/shared/fuses/f_plunging.lua:92]: local something = 200

File: lua/acf/shared/fuses/f_plunging.lua [hunk 2]
```diff
@@ -121,5 +124,6 @@ function this:GetDisplayConfig()
 		["Arming delay"]	= math.Round(self.Primer, 3) .. " s",
 		["Distance"]		= math.Round(self.Distance / 39.37, 1) .. " m",
 		["Detonation Delay"]  = tostring(math.Round(self.Delay, 3)) .. " s"
+		["Adjusted Penetration"] =
```
Comment (KemGus) [inline lua/acf/shared/fuses/f_plunging.lua:127]: empty?

File: lua/acf/shared/guidances/d_infrared.lua [hunk 3]
```diff
@@ -229,10 +229,18 @@ function this:AcquireLock(missile)
 
 		if absang.p < self.SeekCone and absang.y < self.SeekCone then --Entity is within missile cone
 
-			testang = Heat + (360-(absang.p + absang.y)) --Could do pythagorean stuff but meh, works 98% of time
+			testang = absang.p + absang.y --Could do pythagorean stuff but meh, works 98% of time
```
Comment (KemGus) [inline lua/acf/shared/guidances/d_infrared.lua:232]: not the best name. just call it ang or something like that

File: lua/acf/shared/guidances/d_infrared.lua [hunk 4]
```diff
@@ -180,7 +180,7 @@ function this:AcquireLock(missile)
 
 	local missilePos	= missile:GetPos()
 
-	local bestAng	= 0
+	local bestAng	= 5000000
```
Comment (KemGus) [inline lua/acf/shared/guidances/d_infrared.lua:183]: unknown unit
Please add description explaining it

File: lua/weapons/weapon_ace_stinger/cl_init.lua [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		local thiccness = 2
+		surface.SetDrawColor(255, 120, 0, 255)
+
+		if LaunchAuth then
+			tempcolor = Color(0, 150, 0)
+			surface.SetDrawColor(0, 255, 0, 255)
+			thiccness = 5
+		end
+
+		surface.DrawOutlinedRect(x - 215, y - 215, 215 * 2, 215 * 2, thiccness, Color(255, 120, 0))
```
Comment (KemGus) [inline lua/weapons/weapon_ace_stinger/cl_init.lua:26]: spelling error on  "thickness"

File: lua/weapons/weapon_ace_xm25/cl_init.lua [hunk 6]
```diff
@@ -1 +1,92 @@
...
+
+
+function SWEP:DoDrawCrosshair(x, y)
+	local Zoom = self:GetZoomState()
+	self:DrawScope(Zoom)
+
+	local owner = self:GetOwner()
+	local inaccuracy = math.min(owner:GetVelocity():Length() / owner:GetRunSpeed(), 1)
+	inaccuracy = math.max(inaccuracy, self.Heat / self.HeatMax)
+    local fusedelay = XM25FuseDelay or 0
```
Comment (KemGus) [inline lua/weapons/weapon_ace_xm25/cl_init.lua:27]: inconistent indentation

File: lua/weapons/weapon_ace_xm25/cl_init.lua [hunk 7]
```diff
@@ -1 +1,92 @@
...
+    local fusedist = XM25Distance or 0
+    local Tx = XM25Px or 0
+    local Ty = 0
+
+		if Zoom then
```
Comment (KemGus) [inline lua/weapons/weapon_ace_xm25/cl_init.lua:32]: indentation

File: lua/weapons/weapon_ace_xm25/shared.lua [hunk 8]
```diff
@@ -3,6 +3,7 @@ SWEP.Base = "weapon_ace_base"
 SWEP.Category = "ACE Weapons"
 SWEP.SubCategory = "Special"
 SWEP.Purpose = "Lob HE behind cover"
+SWEP.Instructions	= "Left click to shoot. Right click while sprinting to scope, Right click to lase"
```
Comment (KemGus) [inline lua/weapons/weapon_ace_xm25/shared.lua:6]: a bit weird naming for shift. I wonder, if there is a way to return the bind button user has for sprint

**Review by KemGus — CHANGES_REQUESTED** [review]
Just this and linter and this is fine

File: lua/weapons/weapon_ace_xm25/shared.lua (hunk 8 above)
Comment (RedDeadlyCreeper) [reply lua/weapons/weapon_ace_xm25/shared.lua:6]: The function returns when the user is pressing sprint

**Pull request merged**